blocks/modal-header/modal-header.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Decorate the modal header block
+ * Creates a two-column layout with image on left and content on right
+ * If only one column, image takes full width
+ * @param {Element} block the modal header block
+ */
+export default function decorate(block) {
+  const row = block.querySelector(':scope > div');
+  if (!row) return;
+
+  const columns = row.querySelectorAll(':scope > div');
+  if (columns.length === 0) return;
+
+  const imageColumn = columns[0];
+  imageColumn.classList.add('modal-header-image');
+
+  const contentColumn = columns[1];
+  if (contentColumn) {
+    contentColumn.classList.add('modal-header-content');
+  } else {
+    block.classList.add('single-column');
+  }
```

Comment:
im not a big fan of coding columns based on indices.

probably I'd do something like:
- check if cols.lenght ===1 to add single-column class
- loop the columns and check if it has a pacture tag, it it does, add the modal-header-image else add the modal-header-content class